app/routes/languages.rb

```diff
@@ -41,6 +44,23 @@ class Languages < Core
         end
       end
 
+      get '/languages/:track_id/:topic' do |track_id, topic|
+        return language_not_found(topic) unless TOPICS.include?(topic.to_sym)
```

Comment:
(I think you should still be checking the topic exists **after** you have checked the language track exists.)

A topic is clearly not a language, so either:

* the method being called is not named correctly. (this depends on what the method does.)
* It would be better to call a different method such as: `topic_not_found` (you may need to write this.)